source/tyr/tyr/resources.py

```diff
@@ -289,18 +290,27 @@ def get(self, id=None, name=None):
         args = parser.parse_args()
         args.update({'id': id, 'name': name})
         if any(v is not None for v in args.values()):
-            return models.Instance.query.filter_by(**{k: v for k, v in args.items() if v is not None}).all()
+            return models.Instance.query_existing().filter_by(**{k: v for k, v in args.items() if v is not None}).all()
         else:
-            return models.Instance.query.all()
+            return models.Instance.query_existing().all()
 
-    def put(self, id=None, name=None):
-        if id:
-            instance = models.Instance.query.get_or_404(id)
-        elif name:
-            instance = models.Instance.query.filter_by(name=name).first_or_404()
-        else:
-            return ({'error': 'instance is required'}, 400)
+    def delete(self, id=None, name=None):
+        instance = models.Instance.get_from_id_or_name(id, name)
+        if isinstance(instance, tuple):
```

Comment:
in fact, you decide if `instance` is good by testing the type of the returned variable.... @antoine-de  is right, please  manage error cases  of `get_from_id_or_name` by exception...